src/plugins/discover/public/application/main/components/top_nav/on_save_search.tsx

```diff
@@ -101,18 +102,21 @@ export async function onSaveSearch({
     newTitle,
     newCopyOnSave,
     newDescription,
+    newRowsPerPage,
     isTitleDuplicateConfirmed,
     onTitleDuplicate,
   }: {
     newTitle: string;
     newCopyOnSave: boolean;
     newDescription: string;
+    newRowsPerPage?: number;
     isTitleDuplicateConfirmed: boolean;
     onTitleDuplicate: () => void;
   }) => {
     const currentTitle = savedSearch.title;
     savedSearch.title = newTitle;
     savedSearch.description = newDescription;
+    savedSearch.rowsPerPage = newRowsPerPage;
```

Comment:
```suggestion
    savedSearch.rowsPerPage = services.uiSettings.get(DOC_TABLE_LEGACY)
        ? savedSearch.rowsPerPage
        : state.appStateContainer.getState().rowsPerPage;
```

See my comment below for explanation.